proposals/NNNN-unlock-existential-types-for-all-protocols.md

```diff
@@ -120,17 +120,17 @@ extension Animal {
 }
 ```
 
-Contrary to the intent, `Animal` is still assumed as "having an associated type requirement", which happens to be the only thing restraining the existential.
+Unfortunately, `Animal` is still "assumed" to have an associated type requirement, which is solely responsible for restraining the existential.
 
-One more counterproductive aspect to the semantic inconsistency is how authors could be lead to avoid refining useful protocols in fear of their own protocol not qualifying for an existential type.
+The current semantic inconsistency also discourages authors from refining their existing protocols with other, useful ones in fear of losing existential qualification.
 
 ### Library Evolution
 
-Removing the type-level restriction would mean that adding defaulted requirements to a protocol is always both a binary- and source-compatible change, since it couldn't interfere with existing uses of the protocol.
+Removing the type-level restriction would mean that adding defaulted requirements to a protocol is always both a binary- and source-compatible change, since it would not interfere with existing uses of the protocol.
 
 ### Type-erasing Wrappers
 
-For convenience, libraries are often bound to vend custom type-erasing wrappers for commonly used protocols. Since the wrapper must evolve in sync with the protocol, this comes at the expense of resilience, code size and valuable time required to design, document, and maintain these ergonomic interfaces. While removing the restriction doesn't define away the need for manual containers like [`AnySequence`](https://developer.apple.com/documentation/swift/anysequence) and [`AnyHashable`](https://developer.apple.com/documentation/swift/anyhashable), it allows for them to be written in a way that's simpler, easier for the compiler to optimize, and ABI-compatible with future generalized existentials, by wrapping the unconstrained existential type instead of falling back on `Any` or boxing a value in a subclass or closure, and using private protocol extensions to wrap requirements that cannot be directly accessed on the existential:
+For convenience, libraries often vend custom type-erasing wrappers for commonly used protocols. Since the wrapper must evolve in parallel to the protocol, resilience, code size and valuable time required to design, document, and maintain these ergonomic interfaces are all sacrificed. While removing the restriction doesn't define away the need for manual containers, like [`AnySequence`](https://developer.apple.com/documentation/swift/anysequence) and [`AnyHashable`](https://developer.apple.com/documentation/swift/anyhashable), it does allow writing them in a simpler, more easily optmizable, and ABI-compatible way. This is achieved by wrapping the unconstrained existential type instead of falling back on `Any` or boxing a value in a subclass or closure, and using private protocol extensions to wrap requirements that cannot be directly accessed on the existential:
```

Comment:
Why did we choose to loose information on optimizations and ABI compatibility here?

Reply:
Sorry for the optimizations part; is this okay: more easily optimizable __by the compiler__? (I also made a typo optmizable -> optimizable 😅)

As for ABI compatibility I might have misunderstood what you are saying. Could you elaborate?

Reply:
> As for ABI compatibility I might have misunderstood what you are saying. Could you elaborate?

"ABI compatible" by itself has an opaque meaning. It is important to mention that we are talking about binary compatibility with future generalized existentials -- they would allow us to further improve the implementation of some wrappers without breaking the ABI. 

P.S. I borrowed this sentence from Joe's [reply](https://forums.swift.org/t/lifting-the-self-or-associated-type-constraint-on-existentials/18025/26).

Reply:
Perhaps we should include the "explanation" from Joe's reply: "by wrapping the unconstrained existential type instead of boxing a value in a subclass or closure". Also, I think we should either directly quote Joe, or paraphrase so that we don't plagiarize.

Also, generalized existential types are not that well defined, should we include the link to the [relevant section](https://github.com/apple/swift/blob/main/docs/GenericsManifesto.md#generalized-existentials) of the generics manifesto.

Reply:
> Perhaps we should include the "explanation" from Joe's reply: "by wrapping the unconstrained existential type instead of boxing a value in a subclass or closure".

It is already there. I was talking about the deletion of "and ABI-compatible *with future generalized existentials*".

> Also, I think we should either directly quote Joe, or paraphrase so that we don't plagiarize.

Joe can be a co-author, the original pitch was *his* undertaking after all.

> Also, generalized existential types are not that well defined, should we include the link to the relevant section of the generics manifesto.

Cool

Reply:
> It is already there. I was talking about the deletion of "and ABI-compatible with future generalized existentials".
Whoops.

> Joe can be a co-author, the original pitch was his undertaking after all.
Yeah, of course. (Great, I saw that we added him as an author too!)